### PR TITLE
Fix latest Linter violations, kill newly discovered mutation, update corpus, add Gemfile.lock

### PR DESCRIPTION
**Pull request by backus**
I noticed that #72 was failing due to new rubocop violations so I took the time to do some updates

 - Removes one mutation in `Unparser::Commenter` that was introduced
   for #30 but seems to no longer require checking if `node.location`
   exists.

 - Resyncs corpus integrations and adds `repo_ref` from mutant's corpus

 - Add Gemfile.lock for deterministic builds

 - Reduce unit test timeout to 100ms

   The slowest unit tests still run much faster that 100ms
   and this seems to reduce the total runtime of mutant because certain
   mutations don't end up wasintg two seconds a piece.

   Running mutant before this change on my machine:

       Runtime:  209.20s
       Killtime: 959.65s

   After:

       Runtime:  153.32s
       Killtime: 460.45s

 - Update the mutant rake task to use the appropriate amount of jobs
   on circle. Before this was contributing to mutant timing out
   the build

### CODE REVIEW

**Comment by backus** [comment]
Sigh. Also rubyspec issues. I'll resume tomorrow

**Comment by mbj** [comment]
@backus This is another reason we should checkin the `Gemfile.lock` for deterministic builds.

**Comment by backus** [comment]
Yup

File: lib/unparser/constants.rb [hunk 1]
```diff
@@ -27,74 +29,74 @@ def self.symbol_set(enumerable)
...
+    K_OR       = 'or'.freeze
+    K_UNLESS   = 'unless'.freeze
+    K_WHEN     = 'when'.freeze
+    K_WHILE    = 'while'.freeze
+    K_UNTIL    = 'until'.freeze
+    K_YIELD    = 'yield'.freeze
+    K_ENCODING = '__ENCODING__'.freeze
+    K_EEND     = '__END__'.freeze
+    K_FILE     = '__FILE__'.freeze
+    K_THEN     = 'then'.freeze
```
Comment (mbj) [inline lib/unparser/constants.rb]: These changes are redundant with: https://github.com/mbj/unparser/pull/73/files#diff-ecef79bbad605e5acb93fe868716f241R106

Do you really think I left constants unfrozen? OT: as rubocop is not a dynamic tool many of its cops lack precision.

File: lib/unparser/emitter.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Emitter
     extend DSL
 
     # Registry for node emitters
-    REGISTRY = {}
+    REGISTRY = {} # rubocop:disable MutableConstant
```
Comment (mbj) [inline lib/unparser/emitter.rb]: I've a branch to refactor this away BTW.

Comment (mbj) [inline lib/unparser/emitter.rb]: Not a flag for this iteration obviously.

**Comment by mbj** [comment]
@backus I've one real flag here: https://github.com/mbj/unparser/pull/73/files/6b22c70f972991bced614c4660b6c50f373d61e1#diff-ecef79bbad605e5acb93fe868716f241

File: lib/unparser/constants.rb (hunk 1 above)
Comment (backus) [inline lib/unparser/constants.rb]: Ah I didn't see that down below

**Comment by backus** [comment]
Fixed the frozen literals. Will work on porting over the revision locking from mutant's corpus

**Comment by mbj** [comment]
> Fixed the frozen literals. Will work on porting over the revision locking from mutant's corpus

With that one builds should be more deterministic. OT: Should we extract the corpus mechanism to a gem?

**Comment by backus** [comment]
>  OT: Should we extract the corpus mechanism to a gem?

Yeah I think this will help eventually

**Comment by backus** [comment]
Green at last! Ended up doing a lot more project maintenance than I expected! :laughing:

**Comment by mbj** [comment]
> Green at last! Ended up doing a lot more project maintenance than I expected!

At least the tooling forces you to not slip it. And you improved it for the future.

**Pull request merged**